tofu/geom/_core.py

```diff
@@ -5760,6 +5760,86 @@ def calc_min_rho_from_Plasma2D(self, plasma, t=None, log='min',
+                           + " must have kwarg 't=None' for time vector!")
+        is_t_type_valid = (type(t) in [int, float, np.int64, np.float64]
+                           or hasattr(t, '__iter__'))
+        assert t is None or is_t_type_valid, (str_error
+                                              + "Arg t must be None,"
+                                              + " a scalar or an iterable!")
+        # .. Testing outputs ...............................................
+        test_pts = np.array([[1, 2], [3, 4], [5, 6]])
+        npts = test_pts.shape[1]
+        try:
+            out = ff(test_pts, t=t)
+        except Exception:
+            assert False, (str_error
+                           + " must take one positional arg:"
+                           + " a (3, npts) np.ndarray")
+        if hasattr(t, '__iter__'):
+            time_steps = len(t)
+            err_msg = (str_error
+                       + " ff(Pts, t=t), where Pts is a (3, npts) np.array and"
+                       + " t a len()=nt iterable,"
+                       + " must return a (nt, npts) np.ndarray!")
+            assert (type(out) is np.ndarray
+                    and out.shape == (time_steps, npts)), err_msg
+        else:
+            err_msg = (str_error
+                       + " When t=None or t is a scalar,"
+                       + " ff must return a 2D (1, npts) np.ndarray!")
+            assert type(out) is np.ndarray and out.shape == (1, npts), err_msg
+
+        is_ani = ('vect' in kw) if ani is None else ani
+        if is_ani:
+            err_msg = (str_error
+                       + " If ani=True, ff must take a keyword argument:"
+                       + " 'vect=None'!")
+            assert 'vect' in kw, err_msg
+            vect = np.ones(test_pts.shape)
+            try:
+                out = ff(test_pts, vect=vect, t=t)
+            except Exception:
+                err_msg = (str_error
+                           + " If ani=True, ff must handle multiple"
+                           + " points Pts (3, npts) with "
+                           + "multiple vectors (vect as a"
+                           + " (3, npts) np.ndarray)")
+                assert False, err_msg
+            if hasattr(t, '__iter__'):
+                err_msg = (str_error
+                           + " If ani=True, ff must return a (nt, npts)"
+                           + " np.ndarray when Pts is (3,N), vect is provided"
+                           + " and t is a list (nt,)")
+                assert (type(out) is np.ndarray
+                        and out.shape == (time_steps, npts)), err_msg
+            else:
+                msg = (str_error
+                       + "If ani=True, ff must return a (1, npts)"
+                       + " np.ndarray when Pts is (3, npts), vect is"
+                       + " provided and t is None or a scalar")
+                assert type(out) is np.ndarray and out.shape == (1, npts), msg
+        return
```

Comment:
This function should return is_ani

Reply:
Because:
* It used to in the original version
* the default value of ani is None, but the functions called later take ani as input and expect it to be True or False. This function function (check_ff) determines whether ani is True or False, even when it is None. It checks and format ani for other functions called later